### PR TITLE
test: improve spawn_setuid_setgid test

### PR DESCRIPTION
**Pull request by santigimeno**
Check that the child process' uid and gid are correctly set.

### CODE REVIEW

File: test/test-spawn.c [hunk 1]
```diff
@@ -1226,20 +1228,26 @@ TEST_IMPL(spawn_with_an_odd_path) {
 TEST_IMPL(spawn_setuid_setgid) {
   int r;
   struct passwd* pw;
+  char arg2[10];
+  char arg3[10];
```
Comment (cjihrig) [inline test/test-spawn.c]: Maybe name these after the uid and gid.

File: test/run-tests.c [hunk 2]
```diff
@@ -177,5 +178,17 @@ static int maybe_run_test(int argc, char **argv) {
     return spawn_stdin_stdout();
   }
 
+#ifndef _WIN32
+  if (strcmp(argv[1], "spawn_helper_setuid_setgid") == 0) {
+    unsigned int uid = atoi(argv[2]);
```
Comment (cjihrig) [inline test/run-tests.c]: Is it possible to make these of type `uv_uid_t`?

File: test/test-spawn.c [hunk 3]
```diff
@@ -1226,20 +1228,26 @@ TEST_IMPL(spawn_with_an_odd_path) {
...
 
-  init_process_options("spawn_helper1", exit_cb);
+  init_process_options("spawn_helper_setuid_setgid", exit_cb);
 
   /* become the "nobody" user. */
   pw = getpwnam("nobody");
   ASSERT(pw != NULL);
   options.uid = pw->pw_uid;
   options.gid = pw->pw_gid;
+  snprintf(arg2, 10, "%d", pw->pw_uid);
```
Comment (saghul) [inline test/test-spawn.c]: use sizeof instead of 10

**Comment by saghul** [comment]
Not that I dislike the change, but out of curiosity, did you run into any weird problems with the test?

**Comment by santigimeno** [comment]
> Not that I dislike the change, but out of curiosity, did you run into any weird problems with the test?

Not really. I was just reviewing https://github.com/libuv/libuv/pull/938 and saw some issues with the tests in that PR, and thinking about possible solutions: https://github.com/libuv/libuv/pull/938#discussion_r76897643, I thought it was a good idea to implement it for the current tests.

**Comment by saghul** [comment]
:+1:

**Comment by santigimeno** [comment]
PR updated! CI: https://ci.nodejs.org/view/libuv/job/libuv-test-commit/115/

**Comment by cjihrig** [comment]
CI seems "green"

**Comment by santigimeno** [comment]
@saghul @cjihrig LGTY?

**Comment by saghul** [comment]
Oh, sorry, yeah, LGTM

File: test/run-tests.c [hunk 4]
```diff
@@ -177,5 +178,17 @@ static int maybe_run_test(int argc, char **argv) {
     return spawn_stdin_stdout();
   }
 
+#ifndef _WIN32
+  if (strcmp(argv[1], "spawn_helper_setuid_setgid") == 0) {
+    uv_gid_t uid = atoi(argv[2]);
```
Comment (cjihrig) [inline test/run-tests.c]: Should this one be `uv_uid_t`?

**Comment by cjihrig** [comment]
LGTM

File: test/run-tests.c (hunk 4 above)
Comment (santigimeno) [inline test/run-tests.c]: Ouch, you're right.

**Comment by santigimeno** [comment]
Landed as https://github.com/libuv/libuv/commit/4a71e7747a4a2acc888a914033e648e5d8521024

**Pull request closed without merging**